tests/unit/amo/pages/TestAddonReviewList.js

```diff
@@ -839,7 +839,35 @@ describe(__filename, () => {
       );
     });
 
-    it('does not fetchReviewPermissions if they are already loading', () => {
+    it('does not dispatch fetchReviewPermissions() when an error has occured', () => {
+      const addon = { ...fakeAddon };
+      const userId = 66432;
+
+      loadAddon(addon);
+      dispatchSignInActions({ store, userId });
+
+      const id = 'error-handler-id';
+      const error = createApiError({
+        response: { status: 401 },
+        apiURL: 'https://some/api/endpoint',
+        jsonResponse: { message: 'Authentication Failed.' },
+      });
+
+      store.dispatch(setError({ id, error }));
+
+      const capturedError = store.getState().errors[id];
+      // This makes sure the error was dispatched to state correctly.
+      expect(capturedError).toBeTruthy();
+
+      const errorHandler = createStubErrorHandler(capturedError);
```

Comment:
I know we have way too many conflicting patterns in the test suite code but I think using an object directly is better:

```diff
diff --git a/tests/unit/amo/pages/TestAddonReviewList.js b/tests/unit/amo/pages/TestAddonReviewList.js
index 61f74a53a..867280bbb 100644
--- a/tests/unit/amo/pages/TestAddonReviewList.js
+++ b/tests/unit/amo/pages/TestAddonReviewList.js
@@ -16,6 +16,7 @@ import AddonReviewCard from 'amo/components/AddonReviewCard';
 import AddonSummaryCard from 'amo/components/AddonSummaryCard';
 import FeaturedAddonReview from 'amo/components/FeaturedAddonReview';
 import NotFound from 'amo/components/ErrorPage/NotFound';
+import { ErrorHandler } from 'core/errorHandler';
 import Link from 'amo/components/Link';
 import { reviewListURL } from 'amo/reducers/reviews';
 import { DEFAULT_API_PAGE_SIZE, createApiError } from 'core/api';
@@ -846,20 +847,18 @@ describe(__filename, () => {
       loadAddon(addon);
       dispatchSignInActions({ store, userId });
 
-      const id = 'error-handler-id';
       const error = createApiError({
         response: { status: 401 },
         apiURL: 'https://some/api/endpoint',
         jsonResponse: { message: 'Authentication Failed.' },
       });
 
-      store.dispatch(setError({ id, error }));
-
-      const capturedError = store.getState().errors[id];
-      // This makes sure the error was dispatched to state correctly.
-      expect(capturedError).toBeTruthy();
+      const errorHandler = new ErrorHandler({
+        id: 'error-handler-id',
+        dispatch: store.dispatch,
+      });
+      errorHandler.handle(error);
 
-      const errorHandler = createStubErrorHandler(capturedError);
       const dispatchSpy = sinon.spy(store, 'dispatch');
 
       render({ errorHandler });
```

This uses more of a "public" interface for handling the error.

Reply:
ah right, thanks! I reproduced what was in the test file because I could not remember this pattern. I fixed all the test cases in this file :)